packages/app/retainer/commitment.go

```diff
@@ -0,0 +1,97 @@
+package retainer
+
+import (
+	"sync"
+
+	"github.com/iotaledger/goshimmer/packages/core/commitment"
+	"github.com/iotaledger/goshimmer/packages/protocol/ledger/utxo"
+	"github.com/iotaledger/goshimmer/packages/protocol/models"
+	"github.com/iotaledger/hive.go/core/generics/model"
+	"github.com/iotaledger/hive.go/core/serix"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	err := serix.DefaultAPI.RegisterTypeSettings(make(models.BlockIDs, 0), serix.TypeSettings{}.WithLengthPrefixType(serix.LengthPrefixTypeAsUint16))
+	if err != nil {
+		panic(errors.Wrap(err, "error registering BlockIDs type settings"))
+	}
+	err = serix.DefaultAPI.RegisterTypeSettings(utxo.NewTransactionIDs(), serix.TypeSettings{}.WithLengthPrefixType(serix.LengthPrefixTypeAsUint16))
+	if err != nil {
+		panic(errors.Wrap(err, "error registering TransactionIDs type settings"))
+	}
+	err = serix.DefaultAPI.RegisterTypeSettings(utxo.NewOutputIDs(), serix.TypeSettings{}.WithLengthPrefixType(serix.LengthPrefixTypeAsUint16))
+	if err != nil {
+		panic(errors.Wrap(err, "error registering OutputIDs type settings"))
+	}
```

Comment:
These type settings are global. Is this correct? I think you can override the default type settings when you call serialize / deserialize